src/validators/object.ts

```diff
@@ -159,10 +167,52 @@ export abstract class ObjectValidator<T extends ObjectSchema = never, O = never>
         return typeStr
       }
       case 'rust': {
-        throw new Error('Rust not supported yet')
+        if (this.typeGenerated && this.taggedUnionMember) {
+          return `${this.typeName}(${this.typeName}Struct)`
+        }
+        if (this.typeGenerated) {
+          const isOption = !this.required || this.nullable
+          return isOption ? `Option<${this.typeName}>` : `${this.typeName}`
+        }
+
+        const serdeStr = `#[derive(Serialize, Deserialize, Debug, Clone)]\n#[serde(rename_all = "camelCase")]\n`
+        const memberOptions: ValidatorExportOptions = { ...options, parent: this }
+        if (
+          'type' in this.schema &&
```

Comment:
This is a bit to specefic, what if they "tag" is called "key", etc, bascily you would need to pass this down from the Union validator as it can guess the tag or tags.